rebaser.go

```diff
@@ -137,6 +142,71 @@ func (r *Rebaser) validateRebaseable(appImg imgutil.Image) error {
 		if !r.Force && rebaseable == "false" {
 			return fmt.Errorf("app image is not marked as rebaseable")
 		}
+
+		// check the OS, architecture, and variant values
+		// if they are not the same, the image cannot be rebased unless the force flag is set
+		if !r.Force {
+			appOS, err := appImg.OS()
+			if err != nil {
+				return errors.Wrap(err, "get app image os")
+			}
+			newBaseOS, err := newBaseImg.OS()
+			if err != nil {
+				return errors.Wrap(err, "get new base image os")
+			}
+			if appOS != newBaseOS {
+				return fmt.Errorf("incompatible os: '%s' is not compatible with '%s'", newBaseOS, appOS)
+			}
+
+			appArch, err := appImg.Architecture()
+			if err != nil {
+				return errors.Wrap(err, "get app image architecture")
+			}
+			newBaseArch, err := newBaseImg.Architecture()
+			if err != nil {
+				return errors.Wrap(err, "get new base image architecture")
+			}
+			if appArch != newBaseArch {
+				return fmt.Errorf("incompatible architecture: '%s' is not compatible with '%s'", newBaseArch, appArch)
+			}
+
+			appVariant, err := appImg.Variant()
```

Comment:
`platform.TargetMetadata.IsSatisfiedBy()` captures a couple edge cases that we might want to account for (missing variant matches any, etc.).

Reply:
I think I got most of that covered now in f4092ea5 - let me know if you see gaps tho